compiler/frontend-spark/src/main/java/edu/snu/onyx/compiler/frontend/spark/JavaRDD.java

```diff
@@ -0,0 +1,196 @@
+  }
+
+  /**
+   * Constructor.
+   * @param sparkContext spark context containing configurations.
+   * @param parallelism parallelism information.
+   * @param dag the current DAG.
+   * @param lastVertex last vertex added to the builder.
+   */
+  private JavaRDD(final SparkContext sparkContext, final Integer parallelism,
+                  final DAG<IRVertex, IREdge> dag, @Nullable final IRVertex lastVertex) {
+    this.loopVertexStack = new Stack<>();
+    this.sparkContext = sparkContext;
+    this.parallelism = parallelism;
+    this.dag = dag;
+    this.lastVertex = lastVertex;
+    this.kryoSerializer = new KryoSerializer(sparkContext.conf());
+  }
+
+  /////////////// TRANSFORMATIONS ///////////////
+
+  /**
+   * Set initialized source.
+   * @param initialData initial data.
+   * @return the Java RDD with the initialized source vertex.
+   */
+  JavaRDD<T> setSource(final Iterable<T> initialData) {
+    final DAGBuilder<IRVertex, IREdge> builder = new DAGBuilder<>(dag);
+
+    final IRVertex initializedSourceVertex = new InitializedSourceVertex<>(initialData);
+    initializedSourceVertex.setProperty(ParallelismProperty.of(parallelism));
+    builder.addVertex(initializedSourceVertex, loopVertexStack);
+
+    return new JavaRDD<>(this.sparkContext, this.parallelism,
+        builder.buildWithoutSourceSinkCheck(), initializedSourceVertex);
+  }
+
+  /**
+   * Map transform.
+   * @param func function to apply.
+   * @param <O> output type.
+   * @return the JavaRDD with the DAG.
+   */
+  public <O extends Serializable> JavaRDD<O> map(final SerializableFunction<T, O> func) {
+    final DAGBuilder<IRVertex, IREdge> builder = new DAGBuilder<>(dag);
+
+    final IRVertex mapVertex = new OperatorVertex(new MapTransform<>(func));
+    mapVertex.setProperty(ParallelismProperty.of(parallelism));
+    builder.addVertex(mapVertex, loopVertexStack);
+
+    final IREdge newEdge = new IREdge(getEdgeCommunicationPattern(lastVertex, mapVertex),
+        lastVertex, mapVertex, new SparkCoder(kryoSerializer));
+    newEdge.setProperty(KeyExtractorProperty.of(new SparkKeyExtractor()));
+    builder.connectVertices(newEdge);
+
+    return new JavaRDD<>(this.sparkContext, this.parallelism, builder.buildWithoutSourceSinkCheck(), mapVertex);
+  }
+
+  /////////////// ACTIONS ///////////////
+
+  private static final AtomicInteger RESULT_ID = new AtomicInteger(0);
+
+  /**
+   * This method is to be removed after a result handler is implemented.
+   * @return a unique integer.
+   */
+  public static Integer getResultId() {
+    return RESULT_ID.getAndIncrement();
+  }
+
+  /**
+   * Reduce action.
+   * @param func function (binary operator) to apply.
+   * @return the result of the reduce action.
+   */
+  public T reduce(final SerializableBinaryOperator<T> func) {
+    // save result in a temporary file
+    final String resultFile = System.getProperty("user.dir") + "/reduceresult";
```

Comment:
Please leave a comment on why it is needed, and add a TODO if the plan is to remove it in the future.

Reply:
Sure. I'll add a note to make it into a Sink.